logger/redact.go

```diff
@@ -27,35 +27,43 @@ func RedactXML(xml, replacement string, tags []string) string {
 
 // RedactJSON replaces value of key paths from the input JSON with replacement or defaultReplacement when replacement is empty
 //
+// For nested arrays, use `[]` as the key.
+//
 // Example:
 //
-// 		input = `
-// 		{
-// 			"first": "first value",
-// 			"second": {
-// 				"first": "1st of second",
-// 				"second": "2nd of second",
-// 				"third": {
-// 					"first": "1st of second third",
-// 					"second": "2nd of second third",
-// 					"third": "3rd of second third",
-// 				}
-// 			}
-// 		}`
-// 		keys := [][]string{{"first"}, {"second", "second"}, {"second", "third", "first"}}
-// 		RedactJSON(input, "Wego", keys) will return
-// 		{
-// 			"first": "Wego",
-// 			"second": {
-// 				"first": "1st of second",
-// 				"second": "Wego",
-// 				"third": {
-// 					"first": "Wego",
-// 					"second": "2nd of second third",
-// 					"third": "3rd of second third",
-// 				}
-// 			}
-// 		}
+//	input = `
+//	{
+//		"first": "first value",
+//		"second": {
+//			"first": "1st of second",
+//			"second": "2nd of second",
+//			"third": {
+//				"first": "1st of second third",
+//				"second": "2nd of second third",
+//				"third": "3rd of second third",
+//			}
+//		},
+//		"third": [
+//			{ "value": "third value" }
+//		]
+//	}`
+//	keys := [][]string{{"first"}, {"second", "second"}, {"second", "third", "first"}, {"third", "[]", "value"}}
+//	RedactJSON(input, "Wego", keys) will return
+//	{
+//		"first": "Wego",
+//		"second": {
+//			"first": "1st of second",
+//			"second": "Wego",
+//			"third": {
+//				"first": "Wego",
+//				"second": "2nd of second third",
+//				"third": "3rd of second third",
+//			}
+//		},
+//		"third": [
+//			{ "value": "Wego" }
+//		]
+//	}
```

Comment:
same here

Reply:
Done in 59c9ebc. Also tried to make it readable on Go doc using https://github.com/golang/pkgsite:

![image](https://github.com/user-attachments/assets/4b43e2b6-c6e3-4729-9d7e-578d01fd2229)